src/fides/api/service/privacy_request/request_runner_service.py

```diff
@@ -231,6 +231,8 @@ def upload_access_results(  # pylint: disable=R0912
                 privacy_request=privacy_request,
                 data=filtered_results,
                 storage_key=storage_destination.key,  # type: ignore
+                data_category_field_mapping=dataset_graph.data_category_field_mapping,
+                data_use_map=privacy_request.get_cached_data_use_map(),
```

Comment:
So the best I can tell, you're caching a data use map on the privacy request instead of formatting this more similar to dataset_graph.data_category_field_mapping because you need the GraphTask -> Connector -> ConnectionConfig -> System.

If that's the case, can you explain this in the docstring of `_format_data_use_map_for_caching`? Looks like it's not something we can calculate after the fact easily.

Reply:
yeah, well this is more about the graph being subject to change during request execution -  so we want to cache it early on. the issue description mentions this a bit. i'm going to add in some context to the docstring and a code comment here to hopefully make that clearer. let me know if it looks better after that, happy to iterate on this to make sure it's abundantly clear going forward - i think it's probably the least self-explanatory piece of this update 👍

Reply:
ah okay got it